src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -75,16 +108,82 @@ void tryCompileDeepFluentCalls(int depth)
 }");
 
                 var source = builder.ToString();
-
-                var thread = new System.Threading.Thread(() =>
+                RunInThread(() =>
                 {
                     var options = new CSharpCompilationOptions(outputKind: OutputKind.DynamicallyLinkedLibrary, concurrentBuild: false);
                     var compilation = CreateCompilation(source, options: options);
                     compilation.VerifyDiagnostics();
                     compilation.EmitToArray();
-                }, 0);
-                thread.Start();
-                thread.Join();
+                });
+            }
+        }
+
+        [Fact]
+        [WorkItem(33909, "https://github.com/dotnet/roslyn/issues/33909")]
+        public void DeeplyNestedGeneric()
+        {
+            int nestingLevel = (IntPtr.Size * 8) switch
+            {
+                32 when IsDebug => 100,
+                32 when !IsDebug => 100,
+                64 when IsDebug => 100,
+                64 when !IsDebug => 100,
+                _ => throw new Exception($"unexpected pointer size {IntPtr.Size}")
+            };
+
+            // Un-comment loop below and use above commands to figure out the new limits
```

Comment:
Don't forget to comment the loop before merging

Reply:
Blah bad merge.